.github/workflows/teste2e.yml

```diff
@@ -0,0 +1,73 @@
+name: Test e2e
+
+on:
+  pull_request:
+    branches:
+      - develop
+  schedule:
+    - cron: '30 1 * * *'
+  workflow_dispatch:
+    inputs:
+      version_bump:
+        description: 'A type of version bump'
+        default: 'patch'
+        required: true
+        type: choice
+        options:
+          - major
+          - minor
+          - patch
+
+jobs:
+  build-and-deploy-and-test-e2e:
+    runs-on: ubuntu-20.04
+    name: Build & Deploy & Test e2e
+    steps:
+      - uses: actions/checkout@v2
+      - name: Upgrade version
+        run: yarn version --${{ github.event.inputs.version_bump || 'patch' }}  --no-git-tag-version
```

Comment:
why do we need version upgrade there?

Reply:
The reason is, that new commits will be included in a new version, which means that we would like to run tests with a new version. If a test fails, we will know that the new commits, thus the new version is failing. Right now the stable version is v0.0.4, but when I run the e2e test, it makes more sense to see v0.0.5 or v0.1.0 or v1.0.0.

Reply:
Another thing, we don't want to create a git tag and accidentally release a version, that's why I added `--no-git-tag-version`

Reply:
How you can identify that there are new commits by version?
I think that we don't need to bump version temporary. We just need to see commit hash to clearly identify where it is failing

Reply:
Also, I'd suggest to run this test on every PRs.

Reply:
`teste2e.yml` workflow deploys a CF Worker from new commits. If we don't bump the version, CF Worker will have the old version but will not have the same source code. For me, it made more sense to avoid this case. When the test fails, CF Worker's health endpoint will return the stable version number (v0.0.4), instead of the version we actually want the worker to be, such as v0.0.5. So, we will say v0.0.5 is failing, instead of saying v0.0.4 is failing.

> Also, I'd suggest to run this test on every PRs.
That's fair. Let's make it run on every PR. But also, let's make this test pass mandatory only on develop and master, agree?